packages/vitest-pool-workers/src/pool/index.ts

```diff
@@ -372,11 +372,8 @@ function buildProjectWorkerOptions(
 		runnerWorker.compatibilityFlags
 	);
 
-	if (mode !== "v1" && mode !== "v2") {
-		runnerWorker.compatibilityFlags.push(
-			"nodejs_compat",
-			"no_nodejs_compat_v2"
-		);
+	if (mode !== "v2") {
+		runnerWorker.compatibilityFlags.push("nodejs_compat_v2");
 	}
```

Comment:
Could someone clarify this logic here? why are we adding the flag if not present?

Reply:
maybe @petebacondarwin ? This code path is very unfamiliar to me. Worth noting tho that this PR doesn't change the logic by which these flags are applied, so if the original code was doing something wrong I would rather fix that in a followup PR, rather than block this PR on it?